File: app/src/main/java/com/moictab/curriculumvitae/controller/BaseController.java
package com.moictab.curriculumvitae.controller;

import android.content.Context;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;

/**
 * Created by moict on 16/10/2017.
 */

class BaseController {

    private final static String CHARSET = "UTF-8";

    String GetJsonFromResources(Context context, String resourceName) {
        StringBuilder builder = new StringBuilder();
        InputStream json;
        BufferedReader reader;

        try {
            json = context.getAssets().open(resourceName);
            reader = new BufferedReader(new InputStreamReader(json, CHARSET));
            String string;

            while ((string = reader.readLine()) != null) {
                builder.append(string);
            }

            reader.close();
        } catch (Exception ex) {
            ex.printStackTrace();
            return null;
        }

        return builder.toString();
    }

}

File: app/src/main/java/com/moictab/curriculumvitae/controller/BaseController.kt
package com.moictab.curriculumvitae.controller

import android.content.Context
import android.util.Log

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader

open class BaseController {
  fun getJsonFromResources(context: Context, resourceName: String): String? {
    val builder = StringBuilder()
    val json: InputStream
    val reader: BufferedReader

    try {
      json = context.assets.open(resourceName)
      reader = BufferedReader(InputStreamReader(json, CHARSET))
      reader.useLines { lines -> lines.forEach { builder.append(it) } }
      reader.close()
    } catch (e: Exception) {
      Log.e(TAG, "Error getting JSON from assets", e)
    }
    return builder.toString()
  }

  companion object {
    private const val TAG = "BaseController"
    private const val CHARSET = "UTF-8"
  }
}
